watcher/src/bin/main.rs

```diff
@@ -149,14 +152,26 @@ impl WatcherSyncThread {
 
             // For now, ignore origin block, as it does not have a signature.
             let syncing_done = watcher
-                .sync_blocks(1, config.max_block_height, false)
+                .sync_blocks(
+                    1,
+                    config.max_block_height,
+                    Some(max_blocks_per_iteration),
+                    false,
+                )
                 .expect("Could not sync signatures");
-            if syncing_done {
-                log::info!(logger, "sync_signatures indicates we're done");
-                break;
-            }
 
-            sleep(config.poll_interval);
+            // Decide next step before continuing based on sync result
+            match syncing_done {
+                SyncResult::AllBlocksSynced => {
+                    log::info!(logger, "sync_signatures indicates we're done");
+                    break;
```

Comment:
This break seems out of place to me. Shouldn't this watcher thread resume syncing when more blocks can be synced?

Reply:
Not if all blocks are considered synced, the only way this would return True is if the amount of blocks initialized in the config have been synced, in which case the desired behavior is to exit the binary successfully.